scripts/quests/otherAreas/Bombs_Away.lua

```diff
@@ -61,37 +61,9 @@ quest.sections =
             onEventFinish =
             {
                 [8] = function(player, csid, option, npc)
-                    if quest:complete(player) then
-                        player:tradeComplete()
-                    end
-                end,
-            },
-        },
-    },
-    {
-        check = function(player, status, vars)
-            return status == QUEST_COMPLETED
-        end,
-
-        [xi.zone.ULEGUERAND_RANGE] =
-        {
-            ['Buffalostalker_Dodzbraz'] =
-            {
-                onTrigger = function(player, npc)
-                    return quest:event(7)
-                end,
-
-                onTrade = function(player, npc, trade)
-                    if npcUtil.tradeHasExactly(trade, { { xi.items.CLUSTER_CORE, 2 } }) then
-                        return quest:progressEvent(8)
+                    if player:hasCompletedQuest(xi.quest.log_id.OTHER_AREAS, xi.quest.id.otherAreas.BOMBS_AWAY) then
+                        player:delQuest(xi.quest.log_id.OTHER_AREAS, xi.quest.id.otherAreas.BOMBS_AWAY)
```

Comment:
Why are we deleting the quest?  Should we not just be able to put this in a QUEST_COMPLETE section and give the shu'meyo salt if trade:hasExactly(trade, { { xi.items.CLUSTER_CORE, 2 } })?

Reply:
That's what the file was before and it would trigger two rewards, unless I'm misunderstanding?

Reply:
Yeah, this is now basically checking if the quest is unavailable, then deleting the quest when the player completes it.  I'd prefer to see the quest get marked as completed (otherwise it will never show as complete in the player log if we delete it every time).

Reply:
@dallano Can you chime in on why there is a delQuest line here?

Reply:
if the quest is already completed, you can't "recomplete" the quest until it's deleted. So typically it's deleted before completing again.

Reply:
But do we need to recomplete it?  Can it just be granting the award directly?  Just seems odd to delete the quest and re-send a completed status.

Reply:
Tangibly the only thing this changes is the fact that you have to hit "yes" now (with delquest) every time you want to trade him two cluster cores. I'm just going to remove it.